src/doom/info.c

```diff
@@ -1222,6 +1222,7 @@ mobjinfo_t mobjinfo[NUMMOBJTYPES] = {
 	S_PLAY_DIE1,		// deathstate
 	S_PLAY_XDIE1,		// xdeathstate
 	sfx_pldeth,		// deathsound
+	MT_NULL,        // droppeditem
```

Comment:
Two remarks:

1) Manually adding this property to each possible `mobjinfo_t` results in a multi-thousand lines patch that nobody can seriously review.
2) How about the 100 added extra mobj things that are not initialized in this array?

I'd suggest to do the following instead:

Add the `droppeditem` field to the end of the `mobjinfo_t` struct and leave it uninitialized. Then loop over all mobj things and set the property as it fits. This is how I have done it for `actualheight` (i.e. the sprite height in pixels) in `src/doom/p_setup.c:P_InitActualHeights()`. That functions would have to be renamed, though. :wink:

Reply:
Thanks, any suggestions on the new name for `P_InitActualHeights`? maybe `P_InitExtraProperties`?

Reply:
Hmm.. I see P_Init() is called on every level reload, I think changing the properties there might override whatever was set by dehacked.
Or will it not?

Reply:
> Thanks, any suggestions on the new name for `P_InitActualHeights`? maybe `P_InitExtraProperties`?

For example. Or `P_InitThingProperties()`. I don't care that much.



> Hmm.. I see P_Init() is called on every level reload, I think changing the properties there might override whatever was set by dehacked.
> Or will it not?

I think it is only called once in `D_DoomMain`.